xcom2-launcher/xcom2-launcher/Classes/Mod/ModEntry.cs

```diff
@@ -204,12 +274,55 @@ public bool IsInModPath(string modPath)
             return 0 == string.Compare(modPath.TrimEnd('/', '\\'), FilePath.GetDirectoryName(Path), StringComparison.OrdinalIgnoreCase);
         }
 
-		#endregion Mod
+        #endregion Mod
+
+        #region Public Setters
+        public void SetState(ModState newState)
+        {
+            State = newState;
+        }
+
+        public void AddState(ModState newState)
+        {
+            State |= newState;
+        }
+
+        public void RemoveState(ModState newState)
+        {
+            State &= ~newState;
+        }
+
+        public void SetRequiresWOTC(bool NeedWOTC)
+        {
+            BuiltForWOTC = NeedWOTC;
+        }
+
+        public void RealizeSize(long newSize)
+        {
+            Size = newSize;
+        }
+
+        public void SetSource(ModSource newSource)
+        {
+            Source = newSource;
+        }
+
+        public void RealizeIDAndPath(string ModRoot)
+        {
+            Path = ModRoot;
+
+            var info_file = Directory.GetFiles(Path, "*.XComMod", SearchOption.TopDirectoryOnly).SingleOrDefault();
+            if (info_file != null)
+            {
+
```

Comment:
Should probably look like this:
```suggestion
ID = System.IO.Path.GetFileNameWithoutExtension(info_file);
```

Currently RealizeIDAndPath() is only called from DEBUG build, so its not critical but should be fixed anyway.